crates/biome_grit_formatter/tests/specs/grit/marzano/react_hooks.grit.snap

```diff
@@ -0,0 +1,1289 @@
+---
+source: crates/biome_formatter_test/src/snapshot_builder.rs
+info: grit/marzano/react_hooks.grit
+---
+# Input
+
+```grit
+engine marzano(0.1)
+language js
+
+private pattern handle_ref($statements, $statement, $name, $type) {
+  $value where {
+    or {
+      and {
+        or {
+          // The type has an undefined, so our ref type must be undefined
+          and {
+              $statement <: contains js"?",
+              $type <: type_annotation(type=$annotated),
+              $annotated <: not contains js"undefined",
+              $inner_type = js"$annotated | undefined"
+          },
+          // We have a type annotation, use this
+          $type <: type_annotation(type = $inner_type),
+          // Fall back to creating an inner_type this way
+          and {
+              $statement <: contains js"createRef",
+              $statement <: contains or {
+                  type_identifier(),
+                  predefined_type()
+              } as $inner_type
+          }
+        },
+        // We have our type and our ref, so now create the statement
+        $statements += `const $name = useRef<$inner_type>($value);`
+      },
+      and {
+        // Type can't be located, initialize `useRef` without a type
+        $type <: .,
+        $statements += `const $name = useRef($value);`
+      },
+      // We some have a type, so try to use it
+      $statements += `const $name = useRef<$type>($value);`
+    }
+  }
+}
+
+private pattern handle_one_statement($class_name, $statements, $states_statements, $static_statements, $render_statements, $constructor_statements, $handler_callback_suffix, $use_ref_from, $use_previous_value_hook) {
```

Comment:
I suppose here we should break the group on multiple lines. I'll look at it.

It's good to have a complex and long example, we were missing them. Thank you :)